Update to use AMQP job if available.

gauthierm opened the pull request:
https://www.pivotaltracker.com/story/show/83783786

gauthierm committed:
Update parseDuration to be a dynamic method and to take an app context param.

gauthierm committed:
Update media duration to use synchronous AMQP job if available.

charleswaddell commented on Site/dataobjects/SiteAudioMedia.php:
```diff
@@ -74,9 +75,65 @@ public function process($file_path)
 
-	protected function processEncoding($file_path, SiteMediaEncoding $encoding)
+	protected function processEncoding(SiteApplication $app, $file_path,
+		SiteMediaEncoding $encoding)
```
Don't need to pass the `$app` here.

gauthierm committed:
processEncoding() doesn't need the app context param.

charleswaddell commented on Site/SiteMediaDurationUpdater.php:
```diff
@@ -192,7 +192,7 @@ protected function updateDuration(SiteMedia $media)
 		if (file_exists($filename)) {
 			$old_duration = $media->duration;
-			$new_duration = SiteAudioMedia::parseDuration($filename);
+			$new_duration = $media->parseDuration($filename);
```
This method call needs the `$app`.

gauthierm committed:
Pass in application context.

gauthierm committed:
Use correct queue exchange namespace.

gauthierm committed:
Better error handling for JSON decoding and AMQP connection.

gauthierm committed:
Receive message data correctly.

gauthierm committed:
Charles did this better.

charleswaddell committed:
Parse the duration from the first encoding.

We do this because we are unable to parse the
duration from the $file_path file since it isn't
readable by the AMQP server.

charleswaddell committed:
Updated docs.

gauthierm committed:
Merge pull request #2 from charleswaddell/charleswaddell-more-accurate-audio-duration-command

Parse the duration from the first encoding.
